marketplace/google-analytics/src/app.js

```diff
@@ -0,0 +1,135 @@
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    const {auth, parameters, entry} = props;
+    const {prefix, slugId} = parameters;
+    const hasSlug = slugId in entry.fields;
+
+    const pagePath = hasSlug ? `/${prefix ? `${prefix}/` : ''}${entry.fields[slugId].getValue()}/` : '';
+    this.state = {
+      isAuthorized: false,
+      hasSlug,
+      range: {
+        start: new Date(TODAY - DAY_IN_MS * 14),
+        end: TODAY
+      },
+      dimension: 'date',
+      pagePath
+    };
+    auth.on('signIn', () => this.setState({isAuthorized: true}));
+    auth.on('signOut', () => this.setState({isAuthorized: false}));
+    auth.authorize({
+      container: 'auth-button',
+      clientid: CLIENT_ID
+    });
+  }
+
+  handleDateChange({target}) {
+    const {range} = this.state;
+    range[target.name] = target.valueAsDate;
+    this.setState({
+      range
+    });
+  }
+
+  handleDimensionChange({target}) {
+    this.setState({
+      dimension: target.value
+    });
+  }
+
+  render() {
+    const {range, dimension, isAuthorized, pagePath, hasSlug} = this.state;
+    const {auth, entry, parameters} = this.props;
+    if (!isAuthorized) {
+      return <p>Not logged in</p>;
+    }
+
+    if (!hasSlug) {
+      return <p>Slug field is not correctly defined.</p>;
+    }
+
+    if (!entry.getSys().publishedAt) {
+      return <p>Nothing to analyze... entry is not published.</p>;
+    }
+
+    return (
+      <section>
+        <div className="range">
+          <label>
+            From
+            <input
+              name="start"
+              type="date"
+              onChange={this.handleDateChange.bind(this)}
+              value={formatDate(range.start)}
+            />
+          </label>
+          <label>
+            To
+            <input
+              name="end"
+              type="date"
+              onChange={this.handleDateChange.bind(this)}
+              value={formatDate(range.end)}
+              max={formatDate(TODAY)}
+            />
+          </label>
+        </div>
+        <div className="dimensions">
+          {TIMELINE_DIMENSIONS.map(dimension => {
+            const isActive = dimension.value === this.state.dimension;
+            return (
+              <label key={dimension.value} className={isActive ? 'is-active' : ''}>
+                {dimension.label}
+                <input
+                  type="radio"
+                  name="dimension"
+                  value={dimension.value}
+                  onChange={this.handleDimensionChange.bind(this)}
+                  checked={isActive}
+                />
+              </label>
+            );
+          })}
+        </div>
+        <Timeline pagePath={pagePath} range={range} dimension={dimension} viewId={parameters.viewId} />
+        <div className="info">
+          <svg fill="#2D2F31" width="100pt" height="100pt" viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
```

Comment:
Would be nice to extract this icon to a separate component

Reply:
Went ahead and used the Forma 36 component instead :)